xc/common/utils/prjxray_create_place_constraints.py

```diff
@@ -741,6 +753,46 @@ def main():
                 block, vpr_loc, "Constraining clock block {}".format(block)
             )
 
+    # Constrain IDELAYCTRL sites
+    idelayctrl_cmts = set()
+    idelay_instances = place_constraints.get_used_instances("IDELAYE2")
+    for inst in idelay_instances:
+        x, y, z = io_blocks[inst]
+        idelayctrl_cmt = vpr_grid.get_vpr_loc_cmt()[(x, y)]
+        idelayctrl_cmts.add(idelayctrl_cmt)
+
+    idelayctrl_instances = place_constraints.get_used_instances("IDELAYCTRL")
+
+    assert len(idelayctrl_cmts) == len(
+        idelayctrl_instances
+    ), "The number of IDELAYCTRL blocks and IO banks with IDELAYs used do not match."
+
+    idelayctrl_sites = dict()
+    for site_name, _, clk_region in vpr_grid.get_site_type_dict(
+    )['IDELAYCTRL']:
+        if clk_region in idelayctrl_cmts:
+            idelayctrl_sites[clk_region] = site_name
+
+    # Check and remove user constrained IDELAYCTRLs
+    for idelayctrl_block in idelayctrl_instances:
+        if idelayctrl_block in blocks.keys():
+            x, y, _ = blocks[idelayctrl_block]
+            idelayctrl_cmt = vpr_grid.get_vpr_loc_cmt()[(x, y)]
+
+            assert idelayctrl_cmt in idelayctrl_cmts
+
+            idelayctrl_cmts.remove(idelayctrl_cmt)
+            idelayctrl_instances.remove(idelayctrl_block)
+
+    for cmt, idelayctrl_block in zip(idelayctrl_cmts, idelayctrl_instances):
```

Comment:
Should probably add a TODO here about IDELAYCTRL IDELAY_GROUPS?  Also it is worth adding a warning about the fact that `IDELAY_GROUP` parameters are being ignored?  Because this code basically randomly binds IDELAYCTRL's to bank's, which almost certainly not what a user expected.

Reply:
Done